combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/BleDevice.kt

```diff
@@ -159,85 +229,46 @@ internal open class BleDevice (
         })
     }
 
-    fun handleAdvertisement(newAdvertisingData: LegacyProbeAdvertisingData) {
-        advertisingData = newAdvertisingData
-        connectionMonitor.activity()
+    fun observeAdvertisingPackets(serialNumber: String, callback: (suspend (advertisement: CombustionAdvertisingData) -> Unit)? = null) {
+        jobManager.addJob(owner.lifecycleScope.launch(Dispatchers.IO) {
+            DeviceScanner.advertisements.filter {
 
-        // the probe continues to advertise even while a BLE connection is
-        // established.  determine if the device is currently advertising as
-        // connectable or not.
-        val advertisingState = when(newAdvertisingData.isConnectable) {
-            true -> DeviceConnectionState.ADVERTISING_CONNECTABLE
-            else -> DeviceConnectionState.ADVERTISING_NOT_CONNECTABLE
-        }
+                // if advertising packet has same mac address and same probe serial number
+                mac == it.mac && it.probeSerialNumber == serialNumber
```

Comment:
Matt and I talked offline about moving some of this probe-specific functionality closer to where it's used.